pkg/gatewayconfigurationserver/v2/middleware.go

```diff
@@ -12,28 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package gcsv2
+package gatewayconfigurationserver
 
 import (
 	"fmt"
+	"net/http"
 	"strings"
 
-	echo "github.com/labstack/echo/v4"
 	"go.thethings.network/lorawan-stack/v3/pkg/auth"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/webhandlers"
 	"google.golang.org/grpc/metadata"
 )
 
-func (s *Server) normalizeAuthorization(next echo.HandlerFunc) echo.HandlerFunc {
-	return func(c echo.Context) error {
-		ctx := c.Request().Context()
+func (s *Server) fillRequestContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := s.component.FillContext(r.Context())
```

Comment:
The `FillContext` middleware is already on the root router, so this shouldn't be necessary, right?